patches/api/0387-Add-EntityMateEvent.patch

```diff
@@ -0,0 +1,141 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Lulu13022002 <41980282+Lulu13022002@users.noreply.github.com>
+Date: Fri, 24 Jun 2022 11:56:32 +0200
+Subject: [PATCH] Add EntityMateEvent
+
+
+diff --git a/src/main/java/io/papermc/paper/event/entity/EntityMateEvent.java b/src/main/java/io/papermc/paper/event/entity/EntityMateEvent.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..de9a8c84772a343b3ee197a65b587673c80d44b0
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/event/entity/EntityMateEvent.java
+@@ -0,0 +1,129 @@
++package io.papermc.paper.event.entity;
++
++import com.google.common.base.Preconditions;
++import org.bukkit.entity.LivingEntity;
++import org.bukkit.entity.Player;
++import org.bukkit.event.Cancellable;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.entity.EntityEvent;
++import org.bukkit.inventory.ItemStack;
++import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Nullable;
++
++/**
++ * Called when a Turtle or a Frog meets a new mate.
++ */
++public class EntityMateEvent extends EntityEvent implements Cancellable {
```

Comment:
I still think EntityMateEvent and EntityBreedEvent are so similar sounding. Is there a better name we could come up with that refects the difference in mechanics? EntityBreedEvent produces an entity offspring immediately while this causes one of the parent entities to take some action later.

Reply:
I think based on what you've changed on the description we can reuse the word pregnancy for the event name ?
But i agree with you the two names are really similar it's why i have decided to say which entity can trigger this event at the starting point

Reply:
Yeah, I'm not sure. possible words: impregnate, conceive, fertilize. I don't know which one is the best, open to more suggestions.

Reply:
The word fertilize is already used by BlockFertilizeEvent and i think we should avoid to repeat the same name (for two different things) to avoid confusing people. But the other words looks good i let you the choice since i'm better reading english than writing (and have never seen the conceive word)

Reply:
Why not something like EntityFertalizeEggEvent since that's fairly descriptive of what's happening? As far as I could imagine this could cover other animals that may have this mechanic in the future.

Reply:
Yes i like this idea very good to know directly which mechanics that refer

Reply:
Fertalize is not a word, you want Fertilize like you had before.

Reply:
Thank you, renamed